### PR TITLE
docs: added docs for http2

### LINKED ISSUE #810: Write a guide on using HTTP2 in Tailcall
Write a detailed guide on how one can leverage HTTP2 using Tailcall. Talk about both using it on the ingress and the egress side.

**Technical Requirements**
- Add it in docs/guides
- Use excalidraw for imagery.

### PR DESCRIPTION
**Pull request by Min2who**
**Summary:**  
Added docs on leveraging http2 with tailcall

**Issue Reference(s):**  
Fixes #810 
/claim #810

### CODE REVIEW

**Comment by Min2who** [comment]
@tusharmath can yu review it?

File: docs/guides/http2.md [hunk 1]
```diff
@@ -0,0 +1,107 @@
+---
+title: Leveraging HTTP/2 with tailcall
+---
+
+HTTP/2 is a modern protocol that significantly improves web performance by introducing features like multiplexing, header compression, and more efficient connection handling. Tailcall, through its @http operator, enables seamless integration and utilization of HTTP/2 capabilities for both inbound and outbound API requests.
```
Comment (tusharmath) [inline docs/guides/http2.md:5]: FACT: the `@http` operator doesn't affect the inbound requests.

File: docs/guides/http2.md [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+
+## Certificate and Key Generation
```
Comment (tusharmath) [inline docs/guides/http2.md:7]: I prefer a separate guide to setup Tailcall over HTTPS. Also HTTPS should be optional after this :

https://github.com/tailcallhq/tailcall/pull/842

File: docs/guides/http2.md [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+:::Note
+Ensure the cert and key value matches with the file path
+:::
+[getting started]: https://tailcall.run/docs/getting_started/configuration/
+
+- `http`: Specifies the version of HTTP to be used, where `HTTP2` indicates the utilization of the HTTP/2 protocol.
+- `cert`: Points to the path of the certificate file for HTTPS. It's essential for secure communication over HTTP/2.
+- `key`: Refers to the path of the key file needed for HTTP/2. It's vital for secure encryption and decryption of data.
+
+## Egress Side Implementation
```
Comment (tusharmath) [inline docs/guides/http2.md:57]: We should say http2 on egress happens automatically if upstream supports it. Elaborate on that a little bit.

File: docs/guides/http2.md [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+
+```graphql
+type Mutation {
+  createUser(input: UserInput!): User @http(method: "POST", path: "/users", body: "{{args.input}}")
+}
+```
+
+- For methods like POST, use the body field to include data in the request body, substituting variables using Mustache templates.
+
+#### Custom Headers:
```
Comment (tusharmath) [inline docs/guides/http2.md:94]: I find some of this content quite irrelevant to http2.

**Comment by tusharmath** [comment]
@Min2who Are u working on this?

**Comment by Min2who** [comment]
> @Min2who Are u working on this?

yes i am working on it

was outside during the 1st jan

**Comment by Min2who** [comment]
@tusharmath added the changes yu can review it

File: docs/guides/http2.md [hunk 5]
```diff
@@ -0,0 +1,46 @@
+---
+title: Leveraging HTTP/2 with tailcall
+---
+
+HTTP/2 is a modern protocol that significantly improves web performance by introducing features like multiplexing, header compression, and more efficient connection handling. Tailcall, through its @http operator, enables seamless integration and utilization of HTTP/2 capabilities for outbound API requests.
```
Comment (tusharmath) [inline docs/guides/http2.md:5]: Factual Issue: It's not the `@http` operator.

File: docs/guides/http2.md [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+
+Tailcall's @http operator allows seamless integration with HTTP/2 for server-side configurations:
+
+Continuing from the example given in the [getting started] docs
+
+```graphql
+schema @server(port: 8000, graphiql: true, http: HTTP2, cert: "./cert.pem", key: "./key.pem") {
+  query: Query
+  mutation: Mutation
+}
```
Comment (tusharmath) [inline docs/guides/http2.md:17]: Should document why do we need the certs and how to create them.

File: docs/guides/http2.md [hunk 7]
```diff
@@ -0,0 +1,46 @@
...
+```graphql
+type Query {
+  users: [User] @http(path: "/users", baseURL: "https://jsonplaceholder.typicode.com")
+}
+```
+
+- `path`: Specifies the API endpoint for the outgoing request.
+- `baseURL`: Defines the base URL of the API. If omitted, it defaults to the @upstream operator's base URL.
+
+In Tailcall, when you're sending requests to other services (the egress side), it figures out the best way to talk to those services all by itself. So, if the service you're talking to supports HTTP/2, Tailcall just goes ahead and uses that automatically.
```
Comment (tusharmath) [inline docs/guides/http2.md:42]: It does that without any additional configuration.

**Comment by Min2who** [comment]
@tusharmath can yu review it once?

File: docs/guides/http2.md [hunk 8]
```diff
@@ -0,0 +1,52 @@
+---
+title: Leveraging HTTP/2 with tailcall
+---
+
+HTTP/2 is a modern protocol that significantly improves web performance by introducing features like multiplexing, header compression, and more efficient connection handling. Tailcall, through its @http operator, enables seamless integration and utilization of HTTP/2 capabilities for outbound API requests.
```
Comment (tusharmath) [inline docs/guides/http2.md:5]: Tailcall supports HTTP2 on both server and client side ie ingress and egress.

File: docs/guides/http2.md [hunk 9]
```diff
@@ -0,0 +1,52 @@
...
+
+- `http`: Specifies the version of HTTP to be used, where `HTTP2` indicates the utilization of the HTTP/2 protocol.
+- `cert`: Points to the path of the certificate file for HTTPS. It's essential for secure communication over HTTP/2.
+- `key`: Refers to the path of the key file needed for HTTP/2. It's vital for secure encryption and decryption of data.
+
+When you set up Tailcall to use HTTP/2, it's important to include certificates (`cert` and `key`) to make sure communication happens securely via HTTPS. These certificates play a vital role in encrypting and decrypting data, guaranteeing that information remains private and unaltered during its transfer.
+
+- **Creating Certificates**
+
+To generate the necessary certificates (cert.pem and key.pem) for Tailcall's HTTP/2 configuration, you can use tools like [OpenSSL](https://www.openssl.org/source/)
```
Comment (tusharmath) [inline docs/guides/http2.md:33]: Give an example of doing this on mac.

**Comment by tusharmath** [comment]
@Min2who Let's open the PR in https://github.com/tailcallhq/tailcallhq.github.io

**Pull request closed without merging**